llvm/lib/Transforms/InstCombine/InstCombineAndOrXor.cpp

```diff
@@ -4706,6 +4706,13 @@ Instruction *InstCombinerImpl::visitXor(BinaryOperator &I) {
     return Xor;
 
   Value *X, *Y;
+
+  // (A | B) ^ C -> (A ^ C) ^ B
+  if (match(Op0, m_OneUse(m_c_DisjointOr(m_Value(X), m_Value(Y))))) {
```

Comment:
I don't think this will handle `C ^ (A | B)`!

``` suggestion
  if (match(&I, m_c_Xor(m_OneUse(m_c_DisjointOr(m_Value(X), m_Value(Y))), m_Value(M)))) {
 ```
  There's an `M` variable declared above if this is confusing you can declare a variable *Z in the `if` clause
  
  ```
   if (Value *Z; ...)
  ```

Reply:
Thanks for suggestion, i used M